faculty/clients/object.py

```diff
@@ -153,6 +186,32 @@ def list(self, project_id, prefix="/", page_token=None):
             params["pageToken"] = page_token
         return self._get(endpoint, ListObjectsResponseSchema(), params=params)
 
+    def create_directory(self, project_id, path):
+        """Copy objects in the store.
+
+        Parameters
+        ----------
+        project_id : uuid.UUID
+        path : str
+            Create empty object at this source path as a placeholder
+            for a directory
+
+        Raises
+        ------
+        PathAlreadyExists
+            when the path that we want to create as a directory already exists
+        """
+        endpoint = "/project/{}/directory/{}".format(
+            project_id, path.lstrip("/")
+        )
+        try:
+            self._put_raw(endpoint)
+        except Conflict as err:
+            if err.error_code == "object_already_exists":
```

Comment:
What would you think about adding the error_code to the relevant "error" class as a class variable since there's a one-to-one mapping and using it here?

Reply:
In that case we'd probably also want to embed the HTTP error exception class expected as well (i.e. `Conflict`). While the current implementation is not very clever, it is quite readable and I suggest keeping it.